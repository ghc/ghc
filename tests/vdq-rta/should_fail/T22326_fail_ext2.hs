{-# LANGUAGE NoRequiredTypeArguments #-}

module T22326_fail_ext2 where

-- Using visible forall in the type of a term is only possible
-- with the RequiredTypeArguments extension, which is disabled
-- in this module.
f :: forall x -> ()
f = f