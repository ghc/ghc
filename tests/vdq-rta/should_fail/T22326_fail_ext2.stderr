
T22326_fail_ext2.hs:8:6: error: [GHC-51580]
    • Illegal visible, dependent quantification in the type of a term:
        forall {k}. forall (x :: k) -> ()
    • In the type signature: f :: forall x -> ()
    Suggested fix:
      Perhaps you intended to use the ‘RequiredTypeArguments’ extension
